source/mir/random/engine/test.d

```diff
@@ -0,0 +1,232 @@
+module mir.random.engine.test;
+
+import std.typecons;
+import std.stdio;
+
+version(linux) version(X86_64)
+{
+    shared static this()
+    {
+        hasGetRandom = initHasGetRandom;
+    }
+
+    private static bool hasGetRandom;
+
+    auto initHasGetRandom()
+    {
+        import core.sys.posix.sys.utsname;
+        utsname uts;
+        uname(&uts);
+
+        import std.algorithm.iteration : splitter;
+        import std.conv : to;
+
+        auto release = uts.release[].splitter(".");
+        if (release.front.to!int > 3)
+            return true;
+        release.popFront;
+        if (release.front.to!int > 17)
+            return true;
+        return false;
+    }
+
+    /*
+     * Flags for getrandom(2)
+     *
+     * GRND_NONBLOCK	Don't block and return EAGAIN instead
+     * GRND_RANDOM		Use the /dev/random pool instead of /dev/urandom
+     */
+    enum GRND_NONBLOCK = 0x0001;
+    enum GRND_RANDOM = 0x0002;
+    enum GETRANDOM = 318;
+
+    size_t syscall(size_t ident, size_t n, size_t arg1, size_t arg2)
```

Comment:
> Do you mean we can not use C getrandom instead?

It's not part of the C library - it's only available as `syscall`.

> syscall implementation is platform dependent

It's only needed/used for `Linux x86_64`.

> syscall implementation is requires DRuntime because synchronization

Oh - can we do it without `synchronized`?